src/main/java/org/opensearch/search/relevance/transformer/personalizeintelligentranking/reranker/impl/AmazonPersonalizedRankerImpl.java

```diff
@@ -50,6 +56,9 @@ public SearchHits rerank(SearchHits hits, PersonalizeRequestParameters requestPa
                         "response processor configuration or search request parameters");
             }
             List<SearchHit> originalHits = Arrays.asList(hits.getHits());
+            Map<String, Float> idToPersonalizeRankingScoreMap = new HashMap<>();
+            Map<String, Float> idToOpenSearchScoreMap = new HashMap<>();
+            Map<String, SearchHit> itemIdToSearchHitMap = new HashMap<>();
```

Comment:
I would move these maps init after line 93 for combining results 
```
GetPersonalizedRankingResult result = personalizeClient.getPersonalizedRanking(personalizeRequest);
```

Reply:
I have updated it in the revision